src/dg/strong_dg_les.cpp

```diff
@@ -0,0 +1,1076 @@
+#include <deal.II/base/tensor.h>
+#include <deal.II/fe/fe_values.h>
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_tools.h>
+#include <deal.II/dofs/dof_renumbering.h>
+#include <deal.II/dofs/dof_accessor.h>
+#include <deal.II/lac/vector.h>
+#include "ADTypes.hpp"
+#include <deal.II/fe/fe_dgq.h> // Used for flux interpolation
+
+#include "strong_dg_les.hpp"
+
+namespace PHiLiP {
+
+template <int dim, int nstate, typename real, typename MeshType>
+DGStrongLES<dim,nstate,real,MeshType>::DGStrongLES(
+    const Parameters::AllParameters *const parameters_input,
+    const unsigned int degree,
+    const unsigned int max_degree_input,
+    const unsigned int grid_degree_input,
+    const std::shared_ptr<Triangulation> triangulation_input)
+    : DGStrong<dim,nstate,real,MeshType>::DGStrong(parameters_input, degree, max_degree_input, grid_degree_input, triangulation_input)
+{ 
+    if constexpr (dim+2==nstate) {
+        this->pde_model_les_double = std::dynamic_pointer_cast<Physics::LargeEddySimulationBase<dim,dim+2,real>>(this->pde_model_double);
+    }
+}
+
+// Destructor
+template <int dim, int nstate, typename real, typename MeshType>
+DGStrongLES<dim,nstate,real,MeshType>::~DGStrongLES()
+{
+    pcout << "Destructing DGStrongLES..." << std::endl;
+}
+
+template <int dim, int nstate, typename real, typename MeshType>
+void DGStrongLES<dim,nstate,real,MeshType>::allocate_model_variables()
+{
+    // allocate all model variables for each ModelBase object
+    // -- double
+    this->pde_model_double->cellwise_poly_degree.reinit(this->triangulation->n_active_cells(), this->mpi_communicator);
+    this->pde_model_double->cellwise_volume.reinit(this->triangulation->n_active_cells(), this->mpi_communicator);
+}
+
+template <int dim, int nstate, typename real, typename MeshType>
+void DGStrongLES<dim,nstate,real,MeshType>::update_model_variables()
+{
+    // allocate/reinit the model variables
+    allocate_model_variables();
+
+    // TO DO: should only call this once if no hp-adaptation
+    update_cellwise_volume_and_poly_degree();
```

Comment:
for the reviewers: this could be addressed in a future PR -- I only left this comment because it would add some efficiency to the code, but is not currently essential